src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -895,6 +904,37 @@ ModifyQuerySupported(Query *queryTree, Query *originalQuery, bool multiShardQuer
 }
 
 
+/*
+ * Modify statements on simple updetable views are not supported yet.
+ * Actually, we need the original query (the query before postgres
+ * pg_rewrite_query) to detect if the view sitting in rtable is to
+ * be updated or just to be used in FROM clause.
+ * Hence, tracing the postgres source code, we deduced that postgres
+ * puts the relation to be modified to the first entry of rtable.
+ * If first element of the range table list is a simple updatable
+ * view and this view is not coming from FROM clause (inFromCl = False),
+ * then update is run "on" that view.
+ */
+static DeferredErrorMessage *
+DeferErrorIfModifyView(Query *queryTree)
+{
+	RangeTblEntry *firstRangeTableElement = (RangeTblEntry *) lfirst(
```

Comment:
you should use `linitial()`, search Citus code, you'd see the examples of it

Reply:
Replaced `linitial `with `lfirst `in 7128dbe000fcd412d7c51fb17865496cea9322f9